### PR TITLE
add pib (Pigs-In-A-Blanket)

### PR DESCRIPTION
**Pull request by Cpasjuste**

### CODE REVIEW

File: pib/VITABUILD [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+build() {
+  cd "Pigs-In-A-Blanket"
+  make -j$(nproc) USE_VITASDK=1
+}
+
+package () {
+  cd "Pigs-In-A-Blanket"
+  mkdir -p ${pkgdir}${VITASDK}/arm-vita-eabi/lib
+  cp -f ${VITASDK}/arm-vita-eabi/lib/libtaihen_stub.a ${pkgdir}${VITASDK}/arm-vita-eabi/lib/libtaihen_stub.a
+  cp -f ${VITASDK}/arm-vita-eabi/lib/libSceAppMgr_stub.a ${pkgdir}${VITASDK}/arm-vita-eabi/lib/libSceAppMgr_stub.a
```
Comment (d3m3vilurr) [inline pib/VITABUILD:18]: do we need these cp commands?

**Comment by Cpasjuste** [comment]
Hi, thanks this is fixed !

**Comment by Rinnegatamante** [comment]
Will this make sdl2 hard depend on Piglet? If so, that's something that ideally shouldn't be defaulted.

**Comment by Cpasjuste** [comment]
> Will this make sdl2 hard depend on Piglet? If so, that's something that ideally shouldn't be defaulted.

Hi, you'll have to link with pib yes, but the gxm renderer is still available (and default). To use pib/opengl(es) you'll have to init sdl windows with opengl flag. If pib support is disabled, no opengl(es) support is available.

**Comment by isage** [comment]
Wait a little with merging that. I'll remove auto-detection of pib, so by default sdl builds without it. As we discussed on discord.

**Comment by Cpasjuste** [comment]
> Wait a little with merging that. I'll remove auto-detection of pib, so by default sdl builds without it. As we discussed on discord
But it means that we won't be able to use sdl2 + gles (pib) without compiling a custom sdl2 build isn't it ?
I don't see the reason why it would not be enabled by default (you'll have to link a very few more libs).

**Comment by isage** [comment]
We'll provide sdl2-pib and sdl2-vitagl, either as a replacement packages, or with different lib names and separate cmake/pkgconfig configs.
Having pib by default means you can't build with vitagl, as you'll have conflicts.

**Comment by Cpasjuste** [comment]
> We'll provide sdl2-pib and sdl2-vitagl, either as a replacement packages, or with different lib names and separate cmake/pkgconfig configs.
> Having pib by default means you can't build with vitagl, as you'll have conflicts.

Personally I don't see the point to have a vitagl sdl2 (vitagl is a great work, but far from pig which is "native" gles2 calls).

**Comment by Rinnegatamante** [comment]
vitaGL is as "native" as ScePiglet is. I could tell you even several instances in which vitaGL is actually more conformant to GL standards than ScePiglet is (rapid example: gtasa-vita that will not render some objects with ScePiglet due to some non conformant stuffs with uniforms handling in ScePiglet). By the way my point wasn't about vitaGL or Piglet but about avoiding to have an extra hard dependency with base SDL (given how most homebrews will stick to soft rendering and won't use GL context at all).

**Comment by Cpasjuste** [comment]
> vitaGL is as "native" as ScePiglet is. I could tell you even several instances in which vitaGL is actually more conformant to GL standards than ScePiglet is (rapid example: gtasa-vita that will not render some objects with ScePiglet due to some non conformant stuffs with uniforms handling in ScePiglet). By the way my point wasn't about vitaGL or Piglet but about avoiding to have an extra hard dependency with base SDL (given how most homebrews will stick to soft rendering and won't use GL context at all).

I know this is your baby and how much work you must have put in but you seems a little optimistic to me about your implementation :p It's still a fantastic work, no offense..

I still think sdl2+pib should be a standard, but this is not in my hand, I did give my opinion at least !

**Comment by isage** [comment]
> Personally I don't see the point to have a vitagl sdl2. 

That besides point, having pib by default will break linking with vitagl for those, who want it. Even without vitagl support in sdl.

**Comment by Cpasjuste** [comment]
> > Personally I don't see the point to have a vitagl sdl2.
> 
> That besides point, having pib by default will break linking with vitagl for those, who want it. Even without vitagl support in sdl.

It's not possible to link pib and vitagl at the same time? Do you have the time to elaborate on this? (I'm curious why it can't be)

**Comment by isage** [comment]
Yes, because of duplicate gl* symbols.

**Comment by Rinnegatamante** [comment]
Consider that vdpm is supposed to not install every available package at install time meaning those that want GL support can just grab the gl-enabled package whilst those interested in sw rendered only can grab the GL-less package.

**Comment by Rinnegatamante** [comment]
By the way i would move the SDL part of the PR into another PR eventually and leave this PR to pib addition only so it can already be merged.

**Comment by Cpasjuste** [comment]
> By the way i would move the SDL part of the PR into another PR eventually and leave this PR to pib addition only so it can already be merged.

Yep, I'll remove the sdl2 part tomorrow.

**Comment by Cpasjuste** [comment]
Done !

**Review by Rinnegatamante — APPROVED** [review]

**Comment by isage** [comment]
Consider merging #137 first

**Pull request merged**